improvement suggestion feat: log history synthese module

andriacap opened the pull request:
Proposition de travail sur la base des retours de la PR #1835.
Sur l'appui de ce commentaire : 
> Mise à jour du travail restant que nous allons effectuer :
> 
>     * [x]  Supprimer le paramètre LOG_API (superflu)
> 
>     * [ ]  Supprimer l’usage de GenericQuery : la manière fléché par SQLAlchemy pour cela serait d’utiliser query_class ; on peut envisager de faire des fonctions de filtrage générique de cette manière mais il faut que cela soit bien réfléchit et bien couvert par les tests.
> 
>     * [ ]  Supprimer la vue v_log_synthese (plus difficile à maintenir) au profit d’une requête d’union directement dans la route
> 
>     * [ ]  L’UUID est-il vraiment nécessaire ? Si non, ne garder que l’id_synthese
> 
>     * [x]  Renommer le modèle TLogSynthese en SyntheseLogEntry
> 
>     * [ ]  Vérifier la cohérence entre le modèle et le SQL de création de la table (par exemple, supprimer le NOT NULL sur le champs d’UUID dans le SQL)
> 
>     * [ ]  Écrire des tests unitaires
> 
>     * [x]  Fix des conflits et MaJ en v2.11

andriacap committed:
Improving feat: log history synthese module

Improving PR  #1835
According to this comment :
Supprimer l’usage de GenericQuery : la manière fléché par SQLAlchemy pour cela serait d’utiliser query_class ; on peut envisager de faire des fonctions de filtrage générique de cette manière mais il faut que cela soit bien réfléchit et bien couvert par les tests.
Supprimer la vue v_log_synthese (plus difficile à maintenir) au profit d’une requête d’union directement dans la route

Reviewed_by:andriac
[Refs_PR]: #1835

andriacap commented:
Rebonjour @camillemonchicourt ,

J'ai appliqué et testé en local les modifications apportées depuis la branche de @lpofredc (https://github.com/lpofredc/GeoNature/tree/log_history-v2.10) . J'ai du ré adapater les migrations en local car ma base de données était aligné sur la dernière version de geonature . 
J'ai vu que la `down_revision` de la migration `add synthese log history` était lié à une migration bien ancienne à savoir celle ci : `ca0fe5d21ea2` . voir image : 
![image](https://user-images.githubusercontent.com/111564663/216283904-2878e995-61b9-4a21-9463-292eda854aec.png) Je ne sais pas si c'est nécessaire de la lier à cette révision en particulier . 

En attendant , j'attend que @lpofredc  accepte cette PR sur sa branche pour que ça mette à jour la PR #1835 . 

Et si c'est ok , je supprime du coup la PR ([https://github.com/PnX-SI/GeoNature/pull/2311](https://github.com/PnX-SI/GeoNature/pull/2311)) que j'ai créée hier pour ne pas créer de confusion .

bouttier commented:
Il faut effectivement mettre à jour la révision pour que la down_revision point vers la dernière révision de GeoNature présente sur la develop.
Au contraire, conserve ta PR mais met là à jour avec la version de tes travaux basé sur ceux de Fred.

andriacap committed:
feat: log history - change down revision

Change down revision of `add_synthese_log_history` to target head of GN
Branch develop

Reviewed-by: andriacap

andriacap committed:
feat: remove unique_id_sinp- log synthese history

Remove `unique_id_sinp` from `t_log_synthese`
It seems useless to keep ``unique_id_sinp`

Reviewed-by: andriacap

andriacap committed:
feat: removing unused _get_model and get_entity

Remove unused function in _get_model and _get_entity
by usign call directly the model `Synthese`

Reviewed-by: andriacap

camillemonchicourt commented:
PR regroupée sur le dépôt GeoNature - https://github.com/PnX-SI/GeoNature/pull/2337
